src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+from src.openapi_server.models.field_meta_data import FieldMetaData
+from src.openapi_server.models.error_status import ErrorStatus
+
+from odemis.model import _vattributes
+
+_METHOD_GET = 1
+_METHOD_POST = 2
+
+
+class AcquisitionServer(model.HwComponent):
+
+    def __init__(self, name, role, server_URL, children={}, deamon=None, **kwargs):
+        super(AcquisitionServer, self).__init__(name, role, **kwargs)
+
+        self.server_URL = server_URL
+        # TODO K.K. Set external storage and connection
+
+        # Stop any acquisition if already one was in progress
+        self.simpleASMCall(self.server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+
+        # TODO K.K. remove *kwargs and include setup of entire component in try (each sperated) first check if childeren
+        # holds value of component
+        try:
+            kwargs = children["MirrorDescanner"]
+        except Exception:
+            raise ValueError("Required child MirrorDescanner not provided")
+        self._MirrorDescanner = MirrorDescanner("MirrorDescanner", role=None, parent=self)
+        self.children.value.add(self._MirrorDescanner)
+
+        try:
+            kwargs = children["EBeamScanner"]
+        except Exception:
+            raise ValueError("Required child EBeamScanner not provided")
+        self._EBeamScanner = EBeamScanner("EBeamScanner", role=None, parent=self)
+        self.children.value.add(self._EBeamScanner)
+
+        try:
+            kwargs = children["MPPC"]
+        except Exception:
+            raise ValueError("Required child MPPC not provided")
+        self._MPPC = MPPC("MPPC", None, self)
+        self.children.value.add(self._MPPC)
+
+    @classmethod
+    def simpleASMCall(cls, url, method, expected_status, data=None, raw_response=False, timeout=600):
+        """
+
+        :param url: url of the command, server part is defined in global variable URL
+        :param method: getting or posting via global variables _METHOD_GET/_METHOD_POST
+        :param expected_status: expected feedback of server for a positive call
+        :param data: data added to the call
+        :param raw_response: specified the format of the structure returned
+        :param timeout: [s] if within this period no bytes are received an timeout exception is raised
+        :return: status_code(posting), content dictionary(getting), or entire response (raw_response=True)
+        """
+        logging.getLogger().setLevel(logging.INFO)
+        logging.info("Executing: %s" % url)
+        if method == _METHOD_GET:
+            resp = requests.get(url, json=data, timeout=timeout)
+        elif method == _METHOD_POST:
+            resp = requests.post(url, json=data, timeout=timeout)
+
+        if resp.status_code != expected_status:
+            raise AsmApiException(url, resp, expected_status)
+
+        logging.info("Call to %s went fine, no problems occured\n" % url)
```

Comment:
Logging twice is maybe a bit too much?

Reply:
I did this with the idea that if a call is in progress the user can see what is keeping the system occupied. But indeed two logs seems a bit much. If the call succeeded you also like to have a log message (so it doesn't seem like the call is being repeated if multiple cell images are scanned for example).
What do you think?

Reply:
Yes I get your point. I am wondering: If the call does not succeed you should anyways get an error message (raise AsmApiException), don't you? If we are unsure, just keep both now (one before and one after) and we can always get rid of one. Maybe @pieleric  has an opinion on that.

Reply:
@pieleric What do you think? Is having a logging message before the call (indicating what call is being made) and one after (indicating if it succeeded or not) good or do you think only one message is sufficient?

Reply:
Actually, I would say this is *not enough* logging. When something wrong happens you want to know not only which call you made, but also with which argument and what was answered.
So, in first logging, indicate the method, and the data (for some cases, I think the data is going to be really big, so you'll want to remove this later, when the code works fine).
In the second logging, indicate the response code. I think the content might be really big (eg a whole image). So you could log also the first 256 bytes of the content and the length.

Reply:
I limit the output to not overload the error message (Of lists in the data dict only the first 10 values are displayed with a message warning for that).
I only output this when an error will be raised raised so we actually might even leave this in.